File: 백준/Silver/1052. 물병/물병.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());
        int result = 0;

        while(Integer.bitCount(n) > k) {
            result += n & (-n);
            n += n & (-n);
            // n & (-n) : n을 이진수로 표현했을 때 가장 오른쪽에 있는 1
        }

        System.out.println(result);
    }
}

File: 백준/Silver/1052. 물병/물병.kt
fun main() {
    val br = System.`in`.bufferedReader()
    val token = br.readLine().split(" ").map { it.toInt() }
    var n = token[0]
    val k = token[1]
    var result = 0

    while (Integer.bitCount(n) > k) {
        result += n and -n
        n += n and -n
    }
    println(result)
}